src/views/QuestionnaireEdit/IntegerQuestionForm/index.tsx

```diff
@@ -0,0 +1,192 @@
+    useForm,
+    PartialForm,
+} from '@togglecorp/toggle-form';
+
+import {
+    CreateIntegerQuestionMutation,
+    CreateIntegerQuestionMutationVariables,
+    QuestionCreateInput,
+    QuestionTypeEnum,
+} from '#generated/types';
+import IntegerQuestionPreview from '#components/questionPreviews/IntegerQuestionPreview';
+
+import styles from './index.module.css';
+
+const CREATE_INTEGER_QUESTION = gql`
+    mutation CreateIntegerQuestion(
+        $projectId: ID!,
+        $input: QuestionCreateInput!,
+    ){
+        private {
+            projectScope(pk: $projectId) {
+                createQuestion(
+                    data: $input,
+                ) {
+                    ok
+                    errors
+                }
+            }
+        }
+    }
+`;
+
+type FormType = PartialForm<QuestionCreateInput>;
+type FormSchema = ObjectSchema<FormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: () : FormSchemaFields => ({
+        name: {
+            required: true,
+            requiredValidation: requiredStringCondition,
+        },
+        type: {
+            required: true,
+            requiredValidation: requiredStringCondition,
+        },
+        questionnaire: {
+            required: true,
+            requiredValidation: requiredStringCondition,
+        },
+        label: {
+            required: true,
+            requiredValidation: requiredStringCondition,
+        },
+        hint: {},
+    }),
+};
+
+interface Props {
+    projectId: string;
+    questionnaireId: string;
+}
+
+function IntegerQuestionForm(props: Props) {
+    const {
+        projectId,
+        questionnaireId,
+    } = props;
+
+    const alert = useAlert();
+
+    const [
+        triggerQuestionCreate,
+        { loading: createQuestionPending },
+    ] = useMutation<CreateIntegerQuestionMutation, CreateIntegerQuestionMutationVariables>(
+        CREATE_INTEGER_QUESTION,
+        {
+            onCompleted: (questionResponse) => {
+                const response = questionResponse?.private?.projectScope?.createQuestion;
+                if (!response) {
+                    return;
+                }
+                if (response.ok) {
+                    alert.show(
+                        'Question created successfully.',
+                        { variant: 'success' },
+                    );
+                } else {
+                    alert.show(
+                        'Failed to create question.',
+                        { variant: 'error' },
+                    );
+                }
+            },
+            onError: () => {
+                alert.show(
+                    'Failed to create question.',
+                    { variant: 'error' },
+                );
+            },
+        },
+    );
+    const initialFormValue: FormType = {
+        type: 'INTEGER' as QuestionTypeEnum,
+        questionnaire: questionnaireId,
+        name: randomString(),
```

Comment:
I don't think we're supposed to send a random string here. Letting users enter the name for the question does not make sense either, but random string still sounds wrong. What should we do here @AdityaKhatri ?